server/main.go

```diff
@@ -98,13 +99,15 @@ func NameHandler(w http.ResponseWriter, r *http.Request) {
 
 	url := fmt.Sprintf("https://buildpack-registry.s3.amazonaws.com/buildpacks/%s.tgz", id)
 	log.Infof("at=download file=%s url=%s", shimmedBuildpack, url)
-	cmd := fmt.Sprintf(`curl --retry 3 --silent --location "%s" | tar xzm -C %s`, url, target_dir)
-
-	_, err = exec.Command("bash", "-c", cmd).Output()
+	bp, err = downloadBuildpack(url)
+	handlePanic(err)
+	tar := fmt.Sprintf(`tar xzf %s -C %s`, bp, target_dir)
+	_, err = exec.Command("bash", "-c", tar).Output()
 	handlePanic(err)
+	handlePanic(os.Remove(bp))
```

Comment:
I'd like to see `defer os.Remove(bp)` after `downloadBuildpack` to ensure we attempt to remove, even if the tar fails.

Reply:
@jabrown85 this whole dir we're working in gets nuked here: https://github.com/heroku/cnb-shim/blob/master/server/main.go#L66

Reply:
Then why do `os.Remove(bp)` at all?